librz/asm/arch/hexagon/hexagon.h

```diff
@@ -93,33 +75,65 @@ typedef struct {
 } HexOp;
 
 typedef struct {
-	ut32 opcode;
-	ut8 parse_bits;
-	int instruction;
-	ut32 mask;
-	HexPred pred; // Predicate type
-	bool duplex; // is part of duplex container?
-	bool compound; // is part of compound instruction?
-	int shift; // Optional shift left is it true?
-	HexPktInfo pkt_info; // Packet related information. First/last instr., prefix and postfix for mnemonic etc.
-	ut8 op_count;
-	HexOp ops[HEX_MAX_OPERANDS];
-	char mnem_infix[128]; // The mnemonic without the pre- and postfix.
-	char mnem[192]; // Instruction mnemonic
-	ut32 addr; // Memory address the instruction is located.
-	RzAsmOp asm_op;
-	RzAnalysisOp ana_op;
+	bool is_sub; ///< Flag for sub-instructions.
+	ut8 op_count; ///< The number of operands this instruction has.
+	ut32 addr; ///< Memory address the instruction is located (high sub-instruction is unaligned by 2 byte!).
+	ut32 opcode; ///< The instruction opcode.
+	HexPred pred; ///< The instruction predicate.
+	HexInsnID identifier; ///< The instruction identifier
+	char text_infix[128]; ///< Textual disassembly of the instruction.
+	HexOp ops[HEX_MAX_OPERANDS]; ///< The operands of the instructions.
 } HexInsn;
 
+/**
+ * \brief The instruction container holds one instruction or two sub-instructions if it is a duplex.
+ * It stores meta information about those instruction(s) like opcode, packet information or the parse bits.
+ */
+typedef struct {
+	ut8 parse_bits; ///< Parse bits of instruction.
+	bool is_duplex; ///< DOes this container hold two sub-instructions?
+	ut32 identifier; ///< Equals instruction ID if is_duplex = false. Otherwise: (high.id << 16) | (low.id & 0xffff)
+	union {
+		HexInsn *sub[2]; ///< Pointer to sub-instructions if is_duplex = true. sub[0] = high, sub[1] = low
+		HexInsn *insn; ///< Pointer to instruction if is_duplex = false.
+	} bin;
+	ut32 addr; ///< Address of container. Equals address of instruction or of the high sub-instruction if this is a duplex.
+	ut32 opcode; ///< The instruction opcode.
+	HexPktInfo pkt_info; ///< Packet related information. First/last instr., prefix and postfix for text etc.
+	RzAsmOp asm_op; ///< Private copy of AsmOp. Currently only of interest because it holds the utf8 flag.
```

Comment:
Do you plan to expand its use or remove it in the future? In the latter case, I recommend adding `RZ_DEPRECATE`.